ClientGUI/XNAClientJSONLabel.cs

```diff
@@ -0,0 +1,191 @@
+            Logger.Log($"XNAClientJSONLabel [{Name}] processing error: {ex.Message}");
+            WindowManager.AddCallback(() => Text = FallbackText, null);
+        }
+    }
+
+    private string ProcessTemplate(string template, string json)
+    {
+        if (string.IsNullOrEmpty(template))
+        {
+            Logger.Log($"XNAClientJSONLabel [{Name}]: Template is null or empty");
+            return FallbackText;
+        }
+
+        // Regex to match {JSONPath:Format} or {JSONPath}
+        // group 1 = JSONPath, group 2 = format(optional)
+        var regex = new Regex(@"\{([^\}:]+)(?::([^\}]+))?\}");
+
+        var result = regex.Replace(template, match =>
+        {
+            string jsonPath = match.Groups[1].Value.Trim();
+            string format = match.Groups[2].Success ? match.Groups[2].Value.Trim() : null;
+
+            try
+            {
+                var values = ParseJSONPath(json, jsonPath);
+
+                if (values == null || values.Count == 0)
+                    return FallbackText;
+
+                if (MaxResults > 0 && values.Count > MaxResults)
+                    values = values.GetRange(0, MaxResults);
+
+                string value = values.Count == 1 ? values[0] : string.Join(", ", values);
+
+                if (!string.IsNullOrEmpty(format))
+                {
+                    var formatted = new List<string>();
+
+                    foreach (var v in values)
+                    {
+                        if (long.TryParse(v, out long num))
+                            formatted.Add(num.ToString(format));
+                        else if (double.TryParse(v, out double dbl))
+                            formatted.Add(dbl.ToString(format));
+                        else if (DateTime.TryParse(v, out DateTime dt))
+                            formatted.Add(dt.ToString(format));
+                        else
+                            formatted.Add(v);
+                    }
+
+                    return string.Join(", ", formatted);
+                }
+
+                return value;
+            }
+            catch (Exception ex)
+            {
+                Logger.Log($"XNAClientJSONLabel JSONPath '{jsonPath}' error: {ex.Message}");
+                return FallbackText;
+            }
+        });
+
+        return result;
+    }
+    
+    private List<string> ParseJSONPath(string json, string path)
+    {
+        try
+        {
+            var root = Newtonsoft.Json.Linq.JToken.Parse(json);
+            var results = new List<string>();
+
+            var tokens = root.SelectTokens(path);
+
+            if (tokens != null)
+                foreach (var token in tokens)
+                    if (token != null)
+                        results.Add(token.ToString());
+
+            return results.Count > 0 ? results : null;
+        }
+        catch (Exception ex)
+        {
+            Logger.Log($"JSONPath error: {ex.Message}");
+            return null;
+        }
+    }
```

Comment:
I think this should happen way earlier at manager stage, and the control should only receive a parsed json and do template substitution, otherwise we're duplicating work.

Also we already use System.Text.Json, I see that it has no support for JSON Path for some reason, maybe this extension to System.Text.Json would do better? https://www.nuget.org/packages/JsonPath.Net Just so we don't introduce two similar libraries.